### DISCUSSION TITLE
Petition to change the NET6 WinUI project structure

### DISCUSSION THREAD

**Discussion by mikernet** [unoplatform/uno#7435] [Q&A]
I think the multitargeted `Mobile` project is far less ideal than how it was set up previously with individual projects for each platform. This makes it harder to unload a subset of those projects while you are only working on some/one at the time and I'm not a fan of the organization of having subfolders for each platform. That doesn't seem to be an improvement over having that organized into individual projects.

Is there a benefit to doing it this way? Is there even a way I can only build a subset of the targets in the mobile project without editing the `.csproj` file each time? I don't want to wait for iOS builds if I'm currently testing the Android project.

Adding package references to a subset of the targets (or different package versions for different targets) also becomes much more cumbersome as far as I can tell.

**Comment by jeromelaban** [comment]
The current structure is based on the supposed ability of VS 2022 to only build selected target framework. If this continues to be worked on as promised, then the current structure will hold. Otherwise, we'll revisit it.

**Reply by mikernet** [reply]
What is the advantage to organizing it this way, if I may ask? There only seem to be downsides from what I can tell, even with selected target framework only building.

**Reply by jeromelaban** [reply]
It's less clutter and repetition in the solution, for now. You can definitely change the structure in your own project, though. We ultimately want to have everything in the same project, but .NET's limited at this point for this.
